get-started/debug-and-profile.md

```diff
@@ -1,8 +1,174 @@
+
+```
+$ dlv version
+Delve Debugger
+Version: 1.5.0
+Build: $Id: ca5318932770ca063fc9885b4764c30bfaf8a199 $
+```
+
+### Attaching delve to a running TiDB process
+
+As mentioned in [Get the code, build and run](build-tidb-from-source.md), once you get the TiDB server running, you can try to attach the delve debugger.
+
+For example, you can typically build and run a standalone TiDB server by
+
+```
+$ make server
+$ ./bin/tidb-server
+```
+
+in the root directory of the source code.
+
+You can then start a new shell and use `ps` or `pgrep` to find the PID of the tidb server process you just started.
+
+```
+$ pgrep tidb-server
+1394942
+```
+
+If there are multiple PIDs listed, it might be that you have multiple tidb servers running, and you should figure out the PID of the tidb server you are planning to debug with using commands like `ps $PID` where `$PID` is the PID you are trying to know more about.
+
+```
+$ ps 1394942
+    PID TTY      STAT   TIME COMMAND
+1394942 pts/11   SNl    0:02 ./bin/tidb-server
+```
+
+Once you got the PID, you can then attach delve onto it by
+
+```
+$ dlv attach 1394942
+```
+
+You might encounter error messages like
+
+```
+$ dlv attach 1394942
+Could not attach to pid 1394942: this could be caused by a kernel security setting, try writing "0" to /proc/sys/kernel/yama/ptrace_scope
+```
+
+You should follow the instruction and execute the following command as the root user to override the kernel security setting.
+
+```
+# echo 0 > /proc/sys/kernel/yama/ptrace_scope
+```
+
+You should then retry attaching and it should work.
+
+Entering the debugging interface, you will find it familiar if you’ve worked with GDB. It is a interactive dialogue that allows you to interact with the execution of the tidb server attached on. You should type `help` into the dialogue and read the help messages.
+
+### Using delve for debugging
+
+After attaching delve to the running TiDB server process, you can now set breakpoints, and the TiDB server will pause the execution at the breakpoints you specified.
+
+To create a breakpoint, you can write
+
+```
+break [name] <linespec>
+```
+
+where `[name]` stands for the name for the breakpoint, and `<linespec>` is the position of a line of code in the source code.
+
+Once the execution is paused, the context of the execution is fully preserved, and you are free to inspect the value of different variables, print the calling stack and even jump bewteen different goroutines. Once you are satisfied, you can resume the execution by stepping into the next line of code or continue the execution until the next breakpoint is encountered.
+
+Basically, the following steps are typical when you are using debugger:
+
+1. locate the code and set a breakpoint
+2. prepare data so that the execution will get through the breakpoint, and pause there as expected
+3. inspect values, follow the execution step by step
```

Comment:
```suggestion
1. Locate the code and set a breakpoint.
2. Prepare data so that the execution will get through the breakpoint, and pause at the specified breakpoint as expected.
3. Inspect values, and follow the execution step by step.
```